app/routes.py

```diff
@@ -1,2 +1,294 @@
+        # }), 200
+
+    return("PATCH method only. Do not accept other methods for this route.")
+    
+'''
+Wave 05: CRUD for Goal
+'''
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["GET", "POST"])
+
+def handle_goals():
+    if request.method == "GET":
+        query_param_value = request.args.get("sort")
+        if query_param_value == "asc":
+            goals = Task.query.order_by(Goal.title.asc())
+        elif query_param_value == "desc":
+            goals = Task.query.order_by(Goal.title.desc())
+        else:
+            goals= Goal.query.all()
+
+        goals_response = []
+        for goal in goals:
+            goals_response.append(goal.to_dict())
+        return jsonify(goals_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        
+        # Invalid goal if missing title
+        if "title" not in request_body:
+            return jsonify({"details": "Invalid data"}), 400
+        else:
+            new_goal = Goal(
+                title = request_body["title"])
+        
+            db.session.add(new_goal)
+            db.session.commit()
+
+            return make_response({"goal" : new_goal.to_dict()}, 201)
+
+@goals_bp.route("/<goal_id>", methods=["GET","PUT","DELETE"], strict_slashes=False)
+def handle_single_goal(goal_id):
+
+    # Find the goal with the given id
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET": 
+        return make_response({"goal" : goal.to_dict()})
+
+    elif request.method == "PUT":
+        new_data = request.get_json()
+
+        goal.title = new_data["title"]
+
+        db.session.commit()
+
+        return make_response(jsonify(goal=goal.to_dict()), 200)
+    
+    elif request.method == 'DELETE':
+        db.session.delete(goal)
+        db.session.commit()
+
+        return make_response({"details": f"Goal {goal.id} \"{goal.title}\" successfully deleted"}, 200)
+
+'''
+Wave 06: Goal with tasks. POST tasks to a goal. GET tasks under a goal.
+'''
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def post_tasks_to_goal(goal_id):
+
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response("", 404)
+
+    # json body in Postman, what to be posted
+    request_data = request.get_json() 
+        
+    for task_no in request_data["task_ids"]: # see line 26-27 in test. For task_no in task_ids": [1, 4], loop through task_no ([1,4])
+        task = Task.query.get(task_no) # find the task in the Task model that has that id
+        task.goal_id = goal.id # this line creates relationship in db. Assign goal.id from the Goal model to the foreign key in the Task model to the task belongs to that goal. Now that goal has that task. 
+
+    db.session.commit()
+
+    return {
+        "id" : goal.id, 
+        "task_ids" : request_data["task_ids"]
+    }
```

Comment:
This change would have made the above-mentioned bug more obvious. This would also allow the response to include ALL tasks associated with the goal, not just the new ones.
This isn't super intuitive so I'd definitely be down to talk through this more in our next 1:1

```suggestion
    full_task_ids = []
    for task in goal.tasks:
        full_task_ids.append(task.task_id)
        
    return {
        "id" : goal.id, 
        "task_ids" : full_task_ids
    }
```